java/ql/src/semmle/code/java/dataflow/internal/ContainerFlow.qll

```diff
@@ -159,6 +159,41 @@ private predicate taintPreservingArgumentToQualifier(Method method, int arg) {
   method.(CollectionMethod).hasName("offer") and arg = 0
 }
 
+/**
+ * Holds if `method` is a library method that returns tainted data if its
+ * `arg`th argument is tainted.
+ */
+private predicate taintPreservingArgumentToMethod(Method method, int arg) {
+  method.getDeclaringType().hasQualifiedName("java.util", "Collections") and
+  (
+    method
+        .hasName(["singleton", "singletonList", "singletonMap", "enumeration", "list", "max", "min",
```

Comment:
The following appear to be missing:
- `checkedNavigableMap`
- `checkedNavigableSet`
- `newSetFromMap​`
- `synchronizedNavigableMap`
- `synchronizedNavigableSet`
- `unmodifiableNavigableMap​`
- `unmodifiableNavigableSet`

Maybe it would be better to order the method names alphabetically so it is easier to see if a method is contained in the QL set.

Reply:
Thanks, I missed out on the navigable sets and maps. I'll sort the method names.

Reply:
The method `newSetFromMap` is missing on purpose. That method requires the argument to be an empty map so its argument can never be tainted. Of course taint could flow from the returned set back into the wrapped map. However, modelling that is quite a bit of effort for a method that is hardly ever used.